internal/api/rbac/roles.go

```diff
@@ -633,20 +653,20 @@ func ResourcesToRole(
 		slices.Sort(kargoRole.Groups)
 	}
 
-	rules := []rbacv1.PolicyRule{}
+	kargoRole.Rules = []rbacv1.PolicyRule{}
 	for _, role := range roles {
-		rules = append(rules, role.Rules...)
+		kargoRole.Rules = append(kargoRole.Rules, role.Rules...)
 	}
 
-	var err error
-	if kargoRole.Rules, err = NormalizePolicyRules(rules); err != nil {
-		return nil, fmt.Errorf("error normalizing RBAC policy rules: %w", err)
-	}
-
-	if isKargoManaged(sa) &&
-		(len(roles) == 0 || (len(roles) == 1 && isKargoManaged(&roles[0]))) &&
-		(len(rbs) == 0 || (len(rbs) == 1 && isKargoManaged(&rbs[0]))) {
-		kargoRole.KargoManaged = true
+	// Since we cannot make any assumptions that they only contain resource types
+	// we recognize, or that they don't do something really unusual like using a
+	// wildcard resource type, never attempt to normalize rules if any of the
+	// underlying resources are not Kargo-managed.
```

Comment:
Non-Kargo-managed rules are read-only, so it's fine to just show unormalized rules. The whole rule normalization process is really to facilitate easy grant/revoke of specific permissions.